src/sentry/snuba/discover.py

```diff
@@ -1706,12 +1711,33 @@ def find_span_histogram_min_max(span, min_value, max_value, user_query, params,
         calculated_max_value = row[get_function_alias(max_column)]
         max_value = calculated_max_value if calculated_max_value else None
 
-    if max_value is not None and min_value is not None:
-        # min_value may be either queried or provided by the user. max_value was queried.
-        # If min_value > max_value, then max_value should be adjusted with respect to
-        # min_value, since min_value is a lower bound, and any and all data below
-        # min_value should be ignored.
-        max_value = max([max_value, min_value])
+        max_fence_value = None
+        if data_filter == "exclude_outliers":
+            q1_alias = get_function_alias(q1_column)
+            q3_alias = get_function_alias(q3_column)
+
+            first_quartile = row[q1_alias]
+            third_quartile = row[q3_alias]
+
+            if (
+                first_quartile is not None
+                or third_quartile is not None
+                or not math.isnan(first_quartile)
+                or not math.isnan(third_quartile)
+            ):
+                interquartile_range = abs(third_quartile - first_quartile)
+                upper_outer_fence = third_quartile + 3 * interquartile_range
+                max_fence_value = upper_outer_fence
+
+        candidates = [max_fence_value, max_value]
+        candidates = list(filter(lambda v: v is not None, candidates))
+        max_value = min(candidates) if candidates else None
+        if max_value is not None and min_value is not None:
+            # min_value may be either queried or provided by the user. max_value was queried.
+            # If min_value > max_value, then max_value should be adjusted with respect to
+            # min_value, since min_value is a lower bound, and any and all data below
+            # min_value should be ignored.
+            max_value = max([max_value, min_value])
```

Comment:
Just a suggestion, but this is basically duplicated code to the above `find_histogram..`, minus a few lines now that the rest is all implemented and you have the spans calls working nicely. Some copy+paste is fine I think, but I don't really see the histogram outlier function changing much between implementations so it would be nice to make this more util-like and not duplicate it, I think.

Can we do something like pass in a `function_alias` param to the existing code above? 
You'd then have:
```py
q1_alias = get_function_alias(f"{function_alias}({field}, 0.25)")
q3_alias = get_function_alias(f"{function_alias}({field}, 0.75)")
```

```
        functions_acl=["fn_span_exclusive_time"],
```
is probably safe to keep for the base function, it doesn't really matter if it has 1 extra permission.

Reply:
yup, I agree, there's a lot of duplicated code. I'll take a stab at unifying `find_span_histogram_min_max` and `find_histogram_min_max` in a follow up PR 👍